martin/src/srv/tiles.rs

```diff
@@ -100,24 +105,35 @@ impl<'a> DynTileSource<'a> {
             query_str,
             query_obj,
             accept_enc,
+            if_none_match,
             preferred_enc,
             cache,
         })
     }
 
     pub async fn get_http_response(&self, xyz: TileCoord) -> ActixResult<HttpResponse> {
         let tile = self.get_tile_content(xyz).await?;
+        let hash = xxhash_rust::xxh3::xxh3_128(&tile.data);
```

Comment:
In a future PR: plumb the hash from mbtiles and pg through the caching and merging in `get_tile_content` and below.